test/e2e/framework/network/utils.go

```diff
@@ -186,23 +207,29 @@ func (config *NetworkingTestConfig) EndpointHostnames() sets.String {
 //   at least once.
 // - maxTries is the maximum number of curl attempts. If this many attempts pass
 //   and we don't see all expected endpoints, the test fails.
-// - expectedEps is the set of endpointnames to wait for. Typically this is also
-//   the hostname reported by each pod in the service through /hostName.
+// - targetIP is the source Pod IP that will dial the given dialCommand using the given protocol.
+// - dialCommand is the command that the targetIP will send to the targetIP using the given protocol.
+//   the dialCommand should be formatted properly for the protocol (http: URL path+parameters,
+//   udp: command%20parameters, where parameters are optional)
+// - expectedResponses is the unordered set of responses to wait for. The responses are based on
+//   the dialCommand; for example, for the dialCommand "hostname", the expectedResponses
+//   should contain the hostnames reported by each pod in the service through /hostName.
 // maxTries == minTries will confirm that we see the expected endpoints and no
 // more for maxTries. Use this if you want to eg: fail a readiness check on a
 // pod and confirm it doesn't show up as an endpoint.
-func (config *NetworkingTestConfig) DialFromContainer(protocol, containerIP, targetIP string, containerHTTPPort, targetPort, maxTries, minTries int, expectedEps sets.String) {
+func (config *NetworkingTestConfig) DialFromContainer(protocol, dialCommand, containerIP, targetIP string, containerHTTPPort, targetPort, maxTries, minTries int, expectedResponses sets.String) {
 	ipPort := net.JoinHostPort(containerIP, strconv.Itoa(containerHTTPPort))
 	// The current versions of curl included in CentOS and RHEL distros
 	// misinterpret square brackets around IPv6 as globbing, so use the -g
 	// argument to disable globbing to handle the IPv6 case.
-	cmd := fmt.Sprintf("curl -g -q -s 'http://%s/dial?request=hostName&protocol=%s&host=%s&port=%d&tries=1'",
+	cmd := fmt.Sprintf("curl -g -q -s 'http://%s/dial?request=%s&protocol=%s&host=%s&port=%d&tries=1'",
```

Comment:
This is changing from `hostName` -> `hostname`, both are currently handled?

Reply:
ah, I see it here now https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/netexec/netexec.go#L73-L74